packages/tuono-router/src/hooks/useRouter.tsx

```diff
@@ -25,10 +25,7 @@ interface UseRouterHook {
 }
 
 export const useRouter = (): UseRouterHook => {
-  const [location, updateLocation] = useRouterStore((st) => [
-    st.location,
-    st.updateLocation,
-  ])
+  const { location, updateLocation } = useInternalRouter()
 
   const push = (path: string, opt?: PushOptions): void => {
```

Comment:
push should be a stable function. Right now is a new function on every render.
It should be wrapped in a `React.useCallback`

Reply:
Agree